### DISCUSSION TITLE
thread 'main' panicked at 'Unable to find libclang

### DISCUSSION THREAD

**Discussion by chengmou** [filecoin-project/lotus#3967] [General Q&A] [answered]
--- stderr
thread 'main' panicked at 'Unable to find libclang: "couldn\'t find any valid shared libraries matching: [\'libclang.so\', \'libclang-*.so\', \'libclang.so.*\', \'libclang-*.so.*\'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.54.0/src/lib.rs:1959:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

warning: build failed, waiting for other jobs to finish...
error: build failed
+ rm -f /tmp/tmp.DFwSnga90w
make[1]: *** [Makefile:11：.install-filcrypto] 错误 101
make[1]: 离开目录“/root/lotus/extern/filecoin-ffi”
make: *** [Makefile:37：build/.filecoin-install] 错误 2

**Comment by hyunmoon** [comment]
[https://github.com/filecoin-project/lotus/issues/3826](url)

**Comment by jennijuju** [comment, marked as answer]
@chengmou  let me know if you still have this issue after installing llvm and clang by running `sudo apt-get install libclang-dev`

**Comment by jennijuju** [comment]
This seems to be a setup issue, moving to discussion for now, please reopen a ticket if installing the library does not resolve the issue.
